### DISCUSSION TITLE
Phpdoc api

### DISCUSSION THREAD

**Discussion by staabm** [rectorphp/rector#6097] [General]
I am using the new phpdoc ast api in a new custom rector rule. This is a really great thing and an perfect step forwards <3

i have code like
```php
        $variableName = $this->nodeNameResolver->getName($variable);
        if ($variableName === null) {
            throw new \RuntimeException("should not happen");
        }

        // https://github.com/rectorphp/rector/blob/main/docs/how_to_work_with_doc_block_and_comments.md
        $phpDocInfo = $this->phpDocInfoFactory->createFromNodeOrEmpty($statement);

        $found = null;
        foreach ($phpDocInfo->getPhpDocNode()->getVarTagValues() as $varTagValue) {
            if ($varTagValue->variableName == '$' . $variableName) {
                $found = $varTagValue;
                break;
            }
        }
```

I am wondering whether there is a cleaner way instead of doing this string comparisions, when iterating var-tags?

So what I dont like is the `$varTagValue->variableName == '$' . $variableName` thing.
did I miss something?

the complete code can be found at https://github.com/staabm/rector-view-scope/blob/8fdb9c3edb025f2fe1b807f6202871b2d9305791/lib/ViewScopeRector.php#L66

**Comment by TomasVotruba** [comment]
Hi, thanks for feedback :+1: 

This is using original https://github.com/phpstan/phpdoc-parser/blob/master/src/Ast/PhpDoc/VarTagValueNode.php, so your way is correct.
If you have a better idea in mind, try to propose it in repository there. Yet I think this 1:1 to php-parser structure, where public nodes are compared too, so it's quite intuitive to use one way for both packages.

**Reply by staabm** [reply]
I see, thx for the insight.

thinking about it... maybe it would make sense to cover this case with a phpstan rule, which makes sure, that comparison always is used in combination with a `$` str-concat, because otherwise strings will/can never match?

**Reply by staabm** [reply]
just found `getParamTagValueByName` which resolves my initial request

https://github.com/rectorphp/rector-src/blob/db00e23b5e487443b74e612069af9e9a178e45f1/packages/BetterPhpDocParser/PhpDocInfo/PhpDocInfo.php#L398-L411
